### PR TITLE
Update UTM codes

### LINKED ISSUE #112: Update all pulumi.com links to use UTM codes
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

### PR DESCRIPTION
**Pull request by desteves**
- Closes #112
- Skips `./fundamentals` and `./archive` folders
- Uses `markdown-link-check` to test links aren't broken
- Adds `make lint` (Currently prints warning, no errors)

### CODE REVIEW

**Review by jkodroff — CHANGES_REQUESTED** [review]
There's too many changes in this PR for me to review at once. Would it be possible to break this into a few sets of smaller changes?

- The links look like they're fixed, which is good
- Some changes are fine and clear improvements, but not related to fixing links
- Some are changes to content I'm not even sure we still want in the repo
- Some changes are wrong (overly aggressive lint rules) or not related to just fixing links

If possible, could we start with a single MD lint rule just to check that links have UTM codes, run that, and then add in the rest of the changes?

In the meanwhile, we really should do an inventory of our workshop stuff and get specific with what we should propose to retire and what should maybe be graduated into a self-guided tutorial.

**Comment by desteves** [comment]
> Would it be possible to break this into a few sets of smaller changes?

Yes, given the many edits involved, I was debating this myself. I'll break it down.

> If possible, could we start with a single MD lint rule just to check that links have UTM codes, run that, and then add in the rest of the changes?

That works.

File: docker-pulumi/solutions/hello-dbc-done/index.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@
 //   !! You *must* complete the DBC builder setup steps @ https://docs.docker.com/build/cloud/setup/#steps
 // - Docker Desktop / CLI
 // - Pulumi CLI (https://www.pulumi.com/docs/get-started/install/)
-// - *Recommended* Pulumi Cloud account (https://app.pulumi.com/signup)
+// - *Recommended* Pulumi Cloud account (https://app.pulumi.com/signup?utm_source=GitHub&utm_medium=referral&utm_campaign=workshops)
```
Comment (jkodroff) [inline docker-pulumi/solutions/hello-dbc-done/index.ts:17]: Per previous comment, I think we should let the docs be the docs and let the code be the code and refrain from links in comments. (Not relevant to this PR - I know you're just adding the codes where they are needed.)

File: scripts/lint.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  # Check if the file starts with ./archive/ or ./fundamentals
+  if [[ $file == ./archive/* || $file == ./fundamentals* || $file == */node_modules/* ]]; then
+    # echo "Skipping $file"
+    continue
+  fi
+
+  check_utm_code "$file"
+  # Check return code of check_utm_code
+  # if [ $? -ne 0 ]; then
+    # echo "Error: 🚨🚨🚨 Missing UTM code in $file"
```
Comment (jkodroff) [inline scripts/lint.sh:41]: You really do love your emojis.

**Review by jkodroff — APPROVED** [review]
Looks GREAT to me.

File: docker-pulumi/solutions/hello-dbc-done/index.ts (hunk 1 above)
Comment (desteves) [reply docker-pulumi/solutions/hello-dbc-done/index.ts:17]: I'll make this change.

**Pull request merged**